imap/http_dav.c

```diff
@@ -7524,6 +7531,110 @@ static int updates_cb(void *rock, void *data)
     return 0;
 }
 
+/* Callback for reporting collection updates */
+static int sync_col_cb(const mbentry_t *mbentry, void *rock)
+{
+    struct updates_rock *urock = (struct updates_rock *) rock;
+    struct propfind_ctx *fctx = urock->fctx;
+    struct meth_params *rparams = urock->rparams;
+    struct mailbox *mailbox = NULL;
+    int r = 0, code = 0;
+
+    /* Append collection name to URL path */
+    size_t len;
+    char *p;
+
+    if (!fctx->req_tgt->collection) {
+        len = strlen(fctx->req_tgt->path);
+        p = fctx->req_tgt->path + len;
+    }
+    else {
+        p = fctx->req_tgt->collection;
+        len = p - fctx->req_tgt->path;
+    }
+
+    if (p[-1] != '/') {
+        *p++ = '/';
+        len++;
+    }
+
+    mbname_t *mbname = mbname_from_intname(mbentry->name);
+    snprintf(p, MAX_MAILBOX_PATH - len, "%s/",
+             strarray_nth(mbname_boxes(mbname), -1));
+    mbname_free(&mbname);
+
+    /* Skip special calendar collections */
+    if (fctx->req_tgt->namespace->id == URL_NS_CALENDAR &&
```

Comment:
It would be nice to have a flag on these rather than exact name matches, but we have what we have for now!

Reply:
Agreed.  We have flags when parsing a method URL, but not when we're iterating over mailboxes.